base_layer/core/src/blocks/block_header.rs

```diff
@@ -551,4 +508,12 @@ mod test {
         let error_margin = f64::EPSILON; // Use machine epsilon for comparison of floats
         assert!((avg - 60f64).abs() < error_margin);
     }
+
+    #[test]
+    fn block_header_encode_decode() {
+        let header = get_genesis_block(Network::LocalNet).block().header.clone();
+        let header_bytes = header.to_consensus_bytes();
+        let decoded_header = BlockHeader::consensus_decode(&mut header_bytes.as_slice()).unwrap();
+        assert_eq!(header, decoded_header);
+    }
```

Comment:
This also checks that the sized implementation is correct
```suggestion
        let header = get_genesis_block(Network::LocalNet).block().header.clone();
  check_consensus_encoding_correctness(header).unwrap();
    }
```